### ISSUE TITLE
performance issues

### ISSUE THREAD

**Issue by mschop** [mschop/NoTeePHP#8]
@mbj

**Comment by mbj** [comment]
Closed as in no question anymore?

**Comment by mschop** [comment]
@mbj closed because I'm currently creating a more realistic performance test and first would like to test many optimizations I created.

**Comment by mschop** [comment]
@mbj after creating a more realistic test, I'm not facing that big performance issues.
The first test I created were generating 1000 li-elements. While smarty was doing this in 3ms, NoTeePHP took ~10 - ~25ms. In this test I didn't use any complex features of smarty like template inheritance or including other template files.

I now created a new Test where I'm testing a more complex example. I'm doing 1000 generation cycles with this template. Now the difference between Smarty and NoTeePHP are not that big any more.

Smarty:
 │ 2.052 s
 │ 82.69 KiB, 1.44 MiB

NoTeePHP:
 │ 2.871 s
 │ 327.99 KiB, 1.41 MiB

I'm absolutely fine with this result. Most template engines in PHP slower than smarty by a factor of 5.
